diffraction/WISH/bragg-detect/cnn/BraggDetectCNN.py

```diff
@@ -37,28 +47,95 @@ def __init__(self, model_weights_path, batch_size=64, workers=0, iou_threshold=0
         self.iou_threshold = iou_threshold
 
 
-    def find_bragg_peaks(self, workspace, output_ws_name="CNN_Peaks", conf_threshold=0.0, q_tol=0.05):
+    def find_bragg_peaks(self, workspace, output_ws_name="CNN_Peaks", conf_threshold=0.0, **kwargs):
```

Comment:
I would prefer to keep the clustering method as it's own parameter and just pass `**kwargs` to the individual clustering function - I think that would be cleaner? The defaults could live on the individual clustering function declarations?

Reply:
KMeans better be removed I think. 
Do you suggest having a signature like the one below so that `kwargs` can be forwarded to the selected clustering method, each having its default params that would get overridden by the supplied kwargs?
`def find_bragg_peaks(workspace, output_ws_name="CNN_Peaks", conf_threshold=0.0, clustering="QLab", **kwargs)`

Reply:
> KMeans better be removed I think. 

OK, sounds good

> Do you suggest having a signature like the one below so that `kwargs` can be forwarded to the selected clustering method, each having its default params that would get overridden by the supplied kwargs? `def find_bragg_peaks(workspace, output_ws_name="CNN_Peaks", conf_threshold=0.0, clustering="QLab", **kwargs)`

Yes that's my suggestion - if you are happy with this? Maybe you preferred your way for a reason?

Reply:
Yes I see that, I think with the suggested change and e.g.
`_do_hdbscan_clustering(self, peakdata, keep_ignored_labels=True, **kwargs)`
It would behave the same?

Reply:
I think yes, unless we need those kwargs to be supplied to each algorithm class like,
```
hdbscan = HDBSCAN(min_cluster_size=2, 
                          min_samples=2, 
                          store_centers="medoid", 
                          algorithm="auto", 
                          cluster_selection_method="eom", 
                          metric="euclidean", 
                          **kwargs      <---- here
)
```

Reply:
yes I think those `kwargs` would then be passed to e.g. in this case HDBSCAN as you have above